### PR TITLE
feat(NavbarSwitcher): improve env switcher salience

### PR DESCRIPTION
**Pull request by cf-remylenoir**
# Purpose of PR

For our users to easily understand what space and environment they are working in.

- Ticket: https://contentful.atlassian.net/browse/CFISO-2011
- Design: https://www.figma.com/design/RBKfzEzcmiC974nFdtI2wK/Modern-%26-Scalable-%E2%80%93-2024-Webapp-Navigation?node-id=7319-31174&node-type=section&m=dev

## QA

| Before | After |
|--------|--------|
| [Preview](https://5fd1dda724cc620021ace8c5-folfqaowjs.chromatic.com/?path=/story/components-navbar--with-different-environments) | [Preview](https://5fd1dda724cc620021ace8c5-zvjgdlueza.chromatic.com/?path=/story/components-navbar--with-different-environments) | 

## PR Checklist

- [x] I have read the relevant `readme.md` file(s)
- [x] All commits follow our [Git commit message convention](https://github.com/contentful/forma-36/tree/main/packages/forma-36-react-components#commits)
- [x] Tests are added/updated/not required
- [x] Tests are passing
- [x] Storybook stories are added/updated/not required
- [x] Usage notes are added/updated/not required
- [x] Has been tested based on [Contentful's browser support](https://www.contentful.com/faq/about-contentful/#which-browsers-does-contentful-support)
- [x] Doesn't contain any sensitive information

### CODE REVIEW

File: packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx [hunk 1]
```diff
@@ -15,11 +19,28 @@ export function NavbarEnvVariant({
...
-  const color = envVariant === 'master' ? tokens.green600 : tokens.gray600;
+  const isMaster = envVariant === 'master';
+  const color = isMaster ? tokens.green700 : tokens.orange700;
 
-  return isAlias ? (
-    <EnvironmentAliasIcon color={color} className={className} size="medium" />
-  ) : (
-    <EnvironmentIcon color={color} className={className} size="medium" />
-  );
+  switch (true) {
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx:25]: Can we simplify the logic here?

File: packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx [hunk 2]
```diff
@@ -15,11 +19,20 @@ export function NavbarEnvVariant({
...
-  const color = envVariant === 'master' ? tokens.green600 : tokens.gray600;
+  const isMaster = envVariant === 'master';
+  const color = isMaster ? tokens.green700 : tokens.orange700;
 
-  return isAlias ? (
-    <EnvironmentAliasIcon color={color} className={className} size="medium" />
-  ) : (
-    <EnvironmentIcon color={color} className={className} size="medium" />
-  );
+  if (isMaster && !isAlias) {
```
Comment (denkristoffer) [inline packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx:25]: Why are we differentiating between master and aliased to master? For the customer it's the same thing, isn't it?

Comment (cf-remylenoir) [reply packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx:25]: It isn't, we change the true master environment icon to  `RocketLaunchIcon`.

Comment (denkristoffer) [reply packages/components/navbar/src/NavbarSwitcher/NavbarEnvVariant.tsx:25]: What I'm saying is, if an alias is pointing to master, it's essentially master.

**Review by denkristoffer — APPROVED** [review]
For my sake we can approve this now and discuss when Daniel is back.

**Pull request merged**